### ISSUE TITLE
manual_noop_waker makes a suggestion that's too new for my MSRV

### ISSUE THREAD

**Issue by pdh11** [rust-lang/rust-clippy#16849]
### Summary

A recent clippy upgrade started producing new warnings in my crate about "manual implementation of a no-op waker", and suggesting that I should use `std::task::Waker::noop()` instead. But that method was only stabilised in Rust 1.85, and my crate's MSRV is 1.80. Shouldn't clippy refrain from warning about things that my MSRV is too old for me to fix?

### Reproducer

Code:

```rust
pub struct NoOpWaker;

impl Wake for NoOpWaker {
    fn wake(self: Arc<Self>) {}
}
```
(but, importantly, with `rust-version = "1.80"` in `Cargo.toml`)

Current output:

> error: manual implementation of a no-op waker
>   --> cotton-scsi/src/tests/scsi_device.rs:12:6
>    |
> 12 | impl Wake for NoOpWaker {
>    |      ^^^^
>    |
>    = help: use `std::task::Waker::noop()` instead

Desired output:
no warning (or, at least, an alternative suggested fix that rustc 1.80 would accept)

### Version

```text

```

**Comment by samueltardieu** [comment]
> Shouldn't clippy refrain from warning about things that my MSRV is too old for me to fix?

Totally. The fix from @lapla-cogito should be integrated into the next stable release.
